src/nvim/vterm/state.c

```diff
@@ -645,7 +645,7 @@ static int on_escape(const char *bytes, size_t len, void *user)
 
     {
       int setnum = bytes[0] - 0x28;
-      VTermEncoding *newenc = vterm_lookup_encoding(ENC_SINGLE_94, bytes[1]);
+      VTermEncoding *newenc = vterm_lookup_encoding(ENC_SINGLE_94);
 
       if (newenc) {
         state->encoding[setnum].enc = newenc;
```

Comment:
You need to keep the designator, otherwise any `ESC (` sequence will always switch to the DEC graphics mode here (and it will be impossible to switch back).

Another issue here is that I think some applications will switch to DEC graphics char set and then switch back to the "normal" char set with `ESC ( B` (i.e. ASCII). That won't work here now since `ESC ( B` will return NULL from `vterm_lookup_encoding` and the char set will be stuck in the graphics char set.

We could either (1) keep the `usascii` encoding as-is, or (2) just use UTF-8 for the `B` charset.

Reply:
Vote for option (2).

Reply:
I think we might need to keep the usaascii encoding as well as single shift 2 and 3 doesn't work without it, so it's not possible to temporarily switch to another encoding. Let's just keep it as is, I didn't except non-unicode to still be this entrenched in terminal space.